portal-impl/src/com/liferay/portlet/pluginsadmin/action/EditPluginAction.java

```diff
@@ -22,44 +24,47 @@
 import com.liferay.portal.security.auth.PrincipalException;
 import com.liferay.portal.service.PluginSettingServiceUtil;
 import com.liferay.portal.service.PortletServiceUtil;
-import com.liferay.portal.struts.PortletAction;
 import com.liferay.portal.util.PortalUtil;
+import com.liferay.portal.util.PortletKeys;
 
 import java.util.Arrays;
 
 import javax.portlet.ActionRequest;
 import javax.portlet.ActionResponse;
-import javax.portlet.PortletConfig;
-import javax.portlet.RenderRequest;
-import javax.portlet.RenderResponse;
-
-import org.apache.struts.action.ActionForm;
-import org.apache.struts.action.ActionForward;
-import org.apache.struts.action.ActionMapping;
 
 /**
  * @author Brian Wing Shun Chan
  * @author Jorge Ferrer
+ * @author Peter Fellwock
  */
-public class EditPluginAction extends PortletAction {
+
+@OSGiBeanProperties(
+	property = {
+		"javax.portlet.name=" + PortletKeys.PLUGINS_ADMIN,
+		"mvc.command.name=/plugins_admin/edit_plugin"
+	}
+)
+public class EditPluginAction extends BaseMVCActionCommand {
 
 	@Override
-	public void processAction(
-			ActionMapping actionMapping, ActionForm actionForm,
-			PortletConfig portletConfig, ActionRequest actionRequest,
-			ActionResponse actionResponse)
+	protected void doProcessAction(
+			ActionRequest actionRequest, ActionResponse actionResponse)
 		throws Exception {
 
 		try {
 			updatePluginSetting(actionRequest);
 
-			sendRedirect(actionRequest, actionResponse);
+			String redirect = ParamUtil.getString(actionRequest, "redirect");
```

Comment:
Hey @peterfellwock 
I just happened to see this code. This line shouldn't be needed because this should be automatically handled by the LiferayPortlet.java class around line 100. It automatically takes care of the redirect parameter.